src/sidebar/pin/flow/pinFlow.ts

```diff
@@ -38,6 +49,131 @@ function unpinItem(item: TreeItem) {
     PinTreeDataProvider.refresh();
 }
 
+async function startPinScreenFlow() {
+    // Check if user is logged, fail if not so
+    if (!Session.isLoggedIn()) {
+        showNotLoggedInError();
+        return;
+    }
+
+    const savedProjects = getSavedBarrels().filter(barrel => barrel.type === BarrelType.Project);
+
+    // Fail if there is no saved projects
+    if (!savedProjects.length) {
+        MessageBuilder.with(localization.sidebar.pin.noProjectFound)
+            .addOption(localization.common.barrel.add(BarrelType.Project), startAddProjectToSidebarFlow)
+            .addOption(localization.common.cancel)
+            .show();
+        return;
+    }
+
+    // Check if there is only one saved project, show project picker if not so
+    let project = savedProjects.length === 1 ? savedProjects[0] : undefined;
+    if (!project) {
+        // Show project picker
+        const projectQuickPickProvider = new QuickPickProvider(
+            new StaticStore(savedProjects),
+            item => ({
+                label: item.name,
+                detail: getBarrelDetailRepresentation(item)
+            })
+        );
+        projectQuickPickProvider.get().title = localization.sidebar.pin.pinScreen;
+        projectQuickPickProvider.get().placeholder = localization.sidebar.pin.selectProject;
+        project = await projectQuickPickProvider.startSingleSelection();
+    }
+
+    // Fail if no project is selected
+    if (!project) {
+        return;
+    }
+
+    // Show screen picker
+    const screenQuickPickProvider = new QuickPickProvider(
+        new PinnableScreensStore(project.id),
+        item => ({
+            label: item.name,
+            description: item.description
+        })
+    );
+    screenQuickPickProvider.get().title = localization.sidebar.pin.pinScreen;
+    screenQuickPickProvider.get().placeholder = localization.sidebar.pin.selectScreen;
+    const screen = await screenQuickPickProvider.startSingleSelection();
+
+    // Fail if no screen is selected
+    if (!screen) {
+        return;
+    }
+
+    pinUtil.addScreenToPinnedItems(screen, project);
+    BarrelTreeDataProvider.refresh();
+    PinTreeDataProvider.refresh();
+    // TODO: Highlight item in pinned items
+}
+
+async function startPinComponentFlow() {
+    // Check if user is logged, fail if not so
+    if (!Session.isLoggedIn()) {
+        showNotLoggedInError();
+        return;
+    }
+
+    const savedBarrels = getSavedBarrels();
```

Comment:
You get all barrels here since both project and styleguide barrels could have components, right?

Reply:
yes